enhancements/windows-containers/health-management.md

```diff
@@ -0,0 +1,258 @@
+has been set for the Node, or the ConfigMap specified by the desired version annotation has been created, WMCD will
+configure the Windows services to the specifications listed in relevant the ConfigMap. If an error occurs, an event
+will be generated against the Node object.
+
+When configuring a service, WMCD will ensure that the service exists as specified in the ConfigMap. In order to
+properly do this, it must evaluate the variables listed within the ConfigMap. This will be done using the Node object,
+which can be achieved by searching for a Node with an internal IP equivalent to the IP of the VM, and the powershell
+scripts copied to the VM by WMCO.
+
+On a separate thread, WMCD will be periodically polling the state of Windows services. If there is any change in the
+expected state, the services will be created or modified in order to restore expected state. As multiple threads can
+be modifying Windows service state, the appropriate thread safety procedures must be followed.
+
+### Node de-configuration: WMCO responsibilies
+
+When a user indicates to WMCO that it wishes to remove a BYOH node from the cluster, the following procedure will be
+kicked off.
+
+1) WMCO stops the WMCD service on the VM. The dependent services will be automatically stopped as well.
+2) WMCO runs the deconfigure command of WMCD.
+3) WMCO removes the files it copied to the VM.
+
+### Node de-configuration: WMCD responsibilities
+
+When the cleanup command is ran, WMCD will remove the services listed in the ConfigMap used to configure the VM.
+It will then delete the VM's Node object.
+
+### User Stories
+
+User stories can be found within the [health management epic](https://issues.redhat.com/browse/WINC-657).
+
+### API Extensions
+
+N/A
+
+### Risks and Mitigations
+
+* If WMCD has an unrecoverable error that results in the process continuously crashing, the services that it manages
+  will go down, due to them being set as dependents. With the kubelet down, the Node will be unreachable and the Ready
+  status will reflect this. If the VMs are created through the machine-api, the user should set up MachineHealthChecks,
+  which will correct the state by recreating the Machine. For BYOH VMs, user intervention is required at this point.
+  Events and alerts will ensure that an admin is made aware of the issue.
+* There is a possibility that a cluster admin could mistakenly remove the services ConfigMap for an older WMCO version,
+  before Node upgrade is completed. This would result in WMCD not knowing what services it has configured on the VM.
+  In order to prevent this, WMCD can set the Windows service description to "openshift-managed", so that it
+  can search for services and remove them. In order to facilitate upgrades from a version without WMCD, this string
+  should be added to all existing services in 4.10, one major version ahead of WMCD's target release.
```

Comment:
The config map will contain all the services names. Can't we iterate them and stop all the services instead of adding "openshift-managed" description to services? adding them in 4.10 for future release seems unnecessary to me.